fix(validator): identifierScope BUILTIN_AMBIENTS に generic-definitions 14 kind prefix を追加 (#1269 提案 C 漏れ、household-budget dogfood で検出)

## 背景

[#1269](https://github.com/csilost2001/harmony/issues/1269) 提案 C で **processFlowAntipatternValidator** Check 31 (BROKEN_REFERENCE_MATURITY_AWARE) は 24 prefix 全件に拡張済 (PR #1275 / #1278 で完了)。しかし同 PR の scope 外で **identifierScope** validator の `BUILTIN_AMBIENTS` set は **更新されていない**ため、generic-definitions 系の参照 (例: `@msg.<Name>` / `@validation.<Name>` / `@const.<key>`) が `UNKNOWN_IDENTIFIER` エラーとして誤検出される。

## 再現手順

`examples/household-budget/` (branch `feat/example-household-budget`) の取引登録 process-flow で `ValidationRule.message` に `@msg.AmountRequired` を入れて `npm run validate:samples -- ../examples/household-budget` 実行。

```
[ERROR] [identifierScope] [UNKNOWN_IDENTIFIER] actions[0].steps[0].rules[0].message:
  @msg - @msg がこのスコープで宣言されていません
  (inputs / outputs / outputBinding / ambientVariables / loop item のいずれにも無い)
```

## 期待動作 vs 現状

| | 期待 | 現状 |
|---|---|---|
| \`@msg.AmountRequired\` (generic-definitions/message 参照) | identifierScope は skip (catalog 参照は別 validator が担当) | UNKNOWN_IDENTIFIER エラー |
| \`@validation.AmountPositive\` (generic-definitions/validation-rule 参照) | 同上 | UNKNOWN_IDENTIFIER エラー |
| \`@const.TransactionLimits.maxAmount\` (generic-definitions/constants 参照) | 同上 | UNKNOWN_IDENTIFIER エラー |

## 根本原因

\`frontend/src/schemas/identifierScope.ts:46\` の \`BUILTIN_AMBIENTS\` set は generic-definitions 拡張前から変わっておらず、以下 6 prefix のみ:

\`\`\`ts
const BUILTIN_AMBIENTS = new Set<string>([
  \"fn\", \"now\", \"uuid\", \"secret\", \"conv\", \"ambient\",
]);
\`\`\`

generic-definitions 14 kind の参照 prefix がここに無いため、identifierScope.checkStep が変数として解釈してエラーを出す。

## 提案

[docs/spec/process-flow-prefix-system.md](../blob/main/docs/spec/process-flow-prefix-system.md) §3 で確定した 24 prefix のうち、generic-definitions catalog 参照系を BUILTIN_AMBIENTS に追加する:

- \`msg\` (generic-definitions/message)
- \`validation\` (generic-definitions/validation-rule)
- \`const\` (generic-definitions/constants)
- \`event\` (generic-definitions/domain-event、※既存 EventTopic と semantic 同一)
- \`logEvent\` (generic-definitions/log-event)
- \`logConfig\` (generic-definitions/log-config)
- \`screen\` (Screen entity 参照、階層形式 \`@screen.<id>.item.<id>.<field>\`)
- \`table\` (Table entity 参照、階層形式 \`@table.<id>.field.<id>.<field>\`)
- \`view\` (View entity 参照)
- \`viewer\` (ViewDefinition 参照)
- \`layout\` (PageLayout 参照)
- \`fragment\` (generic-definitions/ui-fragment)
- \`behavior\` (generic-definitions/ui-behavior)
- \`contract\` (generic-definitions/data-contract)
- \`type\` (generic-definitions/domain-type)
- \`exception\` (generic-definitions/exception-type)
- \`policy\` (generic-definitions/runtime-policy)
- \`component\` (generic-definitions/component-definition)

これら catalog 参照系の broken-ref 検証は processFlowAntipatternValidator Check 31 が担うため、identifierScope は変数 scope 検査の対象外として skip するだけで OK。

## 影響範囲

- \`frontend/src/schemas/identifierScope.ts\` の BUILTIN_AMBIENTS 拡張 (~20 行追加)
- 対応テスト追加: 各新規 prefix で skip されること確認 (~20 case、または table-driven 1 case)
- 既存 examples は drift しない (該当 prefix を使う既存 flow がない or 既に processFlowAntipatternValidator で warn 扱い)

## 検出経緯

household-budget サンプル (`examples/household-budget/`) を v3 schema から新規構築する dogfood で発見。generic-definitions 14 kind 全件 (37 ファイル) を投入し、process-flow から `@msg.<Name>` で参照しようとした際に identifierScope エラーが発火。本サンプルでは暫定回避策として `@conv.msg.<key>` (旧 conventions 規約参照) に書き換えて validate:samples を 4/4 pass させているが、canonical 設計は `@msg.<Name>` であるため本 ISSUE で根治。

## 関連

- 親: Refs [#1269](https://github.com/csilost2001/harmony/issues/1269) 提案 C (closed、processFlowAntipatternValidator 側は完了済)
- spec: [docs/spec/process-flow-prefix-system.md](../blob/main/docs/spec/process-flow-prefix-system.md) §3 (24 prefix list canonical)
- 検出 PR (未マージ): \`feat/example-household-budget\` branch (household-budget サンプル新規構築の 3 commit)

--- a/frontend/src/schemas/identifierScope.test.ts
+++ b/frontend/src/schemas/identifierScope.test.ts
@@ -387,6 +387,64 @@ describe("checkIdentifierScopes — 組み込み関数 BUILTIN_AMBIENTS", () => 
     }));
     expect(issues.some((i) => i.identifier === "reallyUnknownVar")).toBe(true);
   });
+
+  // #1285: Generic Definition Catalog refs + Top-level entity hierarchical refs を
+  // 一括で table-driven テスト。catalog 参照系の broken-ref は別 validator
+  // (processFlowAntipatternValidator Check 31) が担うため identifierScope は skip するだけで OK。
+  describe.each([
+    // Generic Definition Catalog (13 prefix)
+    ["msg",        "@msg.AmountRequired"],
+    ["validation", "@validation.AmountPositiveRange"],
+    ["const",      "@const.TransactionLimits.maxAmount"],
+    ["event",      "@event.transaction.created"],
+    ["logEvent",   "@logEvent.TransactionAuditCreated"],
+    ["logConfig",  "@logConfig.DefaultLogConfig"],
+    ["fragment",   "@fragment.CategoryBadge"],
+    ["behavior",   "@behavior.FormDirtyConfirmExit"],
+    ["contract",   "@contract.TransactionCreateRequest"],
+    ["type",       "@type.Money"],
+    ["exception",  "@exception.TransactionNotFoundException"],
+    ["policy",     "@policy.BackendRetryPolicy"],
+    ["component",  "@component.BalanceCalculator"],
+    // Top-level entity hierarchical refs (5 prefix)
+    ["screen",     "@screen.userForm.item.email.value"],
+    ["table",      "@table.users.field.id.value"],
+    ["view",       "@view.v_monthly_summary.field.total_amount"],
+    ["viewer",     "@viewer.transactionListViewer"],
+    ["layout",     "@layout.defaultLayout"],
+  ])("#1285: @%s 参照は UNKNOWN_IDENTIFIER を出さない", (prefix, expression) => {
+    it(`expression="${expression}" で suppress される`, () => {
+      const issues = checkIdentifierScopes(makeGroup({
+        actions: [{
+          id: "a1", name: "f", trigger: "click",
+          steps: [
+            { id: "s1", kind: "compute", description: "", expression, outputBinding: "r" },
+          ],
+        }],
+      }));
+      expect(issues.filter((i) => i.identifier === prefix)).toHaveLength(0);
+    });
+  });
+
+  it("#1285: ValidationRule.message 内の @msg.<Name> も検出されない (household-budget dogfood で発見した実シナリオ)", () => {
+    const issues = checkIdentifierScopes(makeGroup({
+      actions: [{
+        id: "a1", name: "f", trigger: "click",
+        inputs: [{ name: "amount", type: "number" }],
+        steps: [
+          {
+            id: "s1", kind: "validation", description: "",
+            fieldErrorsVar: "fieldErrors",
+            rules: [
+              { field: "amount", type: "required", severity: "error", message: "@msg.AmountRequired" },
+              { field: "amount", type: "range", min: 1, max: 100000000, severity: "error", message: "@msg.AmountOutOfRange" },
+            ],
+          },
+        ],
+      }],
+    }));
+    expect(issues.filter((i) => i.identifier === "msg")).toHaveLength(0);
+  });
 });
 
 describe("checkIdentifierScopes - TransactionScopeStep onRollback @error ambient", () => {

--- a/frontend/src/schemas/identifierScope.test.ts
+++ b/frontend/src/schemas/identifierScope.test.ts
@@ -388,13 +388,14 @@ describe("checkIdentifierScopes — 組み込み関数 BUILTIN_AMBIENTS", () => 
     expect(issues.some((i) => i.identifier === "reallyUnknownVar")).toBe(true);
   });
 
-  // #1285: Generic Definition Catalog refs + Top-level entity hierarchical refs を
+  // #1285: Generic Definition Catalog refs + Top-level entity / catalog refs を
   // 一括で table-driven テスト。catalog 参照系の broken-ref は別 validator
   // (processFlowAntipatternValidator Check 31) が担うため identifierScope は skip するだけで OK。
   describe.each([
-    // Generic Definition Catalog (13 prefix)
+    // Generic Definition Catalog (14 kind 全件)
     ["msg",        "@msg.AmountRequired"],
     ["validation", "@validation.AmountPositiveRange"],
+    ["rule",       "@rule.DeficitWarning"],
     ["const",      "@const.TransactionLimits.maxAmount"],
     ["event",      "@event.transaction.created"],
     ["logEvent",   "@logEvent.TransactionAuditCreated"],
@@ -406,12 +407,16 @@ describe("checkIdentifierScopes — 組み込み関数 BUILTIN_AMBIENTS", () => 
     ["exception",  "@exception.TransactionNotFoundException"],
     ["policy",     "@policy.BackendRetryPolicy"],
     ["component",  "@component.BalanceCalculator"],
-    // Top-level entity hierarchical refs (5 prefix)
+    // Top-level entity / catalog refs (9 prefix)
     ["screen",     "@screen.userForm.item.email.value"],
     ["table",      "@table.users.field.id.value"],
     ["view",       "@view.v_monthly_summary.field.total_amount"],
     ["viewer",     "@viewer.transactionListViewer"],
     ["layout",     "@layout.defaultLayout"],
+    ["seq",        "@seq.orderNumberSeq"],
+    ["flow",       "@flow.createOrder"],
+    ["system",     "@system.stripeApi"],
+    ["ext",        "@ext.retail"],
   ])("#1285: @%s 参照は UNKNOWN_IDENTIFIER を出さない", (prefix, expression) => {
     it(`expression="${expression}" で suppress される`, () => {
       const issues = checkIdentifierScopes(makeGroup({

--- a/frontend/src/schemas/identifierScope.ts
+++ b/frontend/src/schemas/identifierScope.ts
@@ -42,10 +42,11 @@ export interface IdentifierIssue {
  * - ambient: @ambient.* 旧形式の ambient 参照
  *
  * **Generic Definition Catalog 参照** (docs/spec/process-flow-prefix-system.md §3、
- * #1285 で追加): broken-ref 検証は processFlowAntipatternValidator Check 31 が担うため
- * identifierScope は変数 scope 検査の対象外として skip する。
+ * #1285 で追加、14 kind 全件): broken-ref 検証は processFlowAntipatternValidator Check 31
+ * が担うため identifierScope は変数 scope 検査の対象外として skip する。
  * - msg        : @msg.<Name> generic-definitions/message
- * - validation : @validation.<Name> generic-definitions/validation-rule
+ * - validation : @validation.<Name> generic-definitions/validation-rule (inline boolean return)
+ * - rule       : @rule.<Name> generic-definitions/application-rule
  * - const      : @const.<Name>.<key> generic-definitions/constants
  * - event      : @event.<topic> generic-definitions/domain-event (catalogs.events と兼用)
  * - logEvent   : @logEvent.<Name> generic-definitions/log-event
@@ -64,6 +65,10 @@ export interface IdentifierIssue {
  * - view   : @view.<id>.field.<col> View entity 参照
  * - viewer : @viewer.<id> ViewDefinition 参照
  * - layout : @layout.<id> PageLayout 参照
+ * - seq    : @seq.<id> Sequence entity 参照
+ * - flow   : @flow.<id> ProcessFlow entity 参照 (inline 禁止、副作用 invocation)
+ * - system : @system.<id> ExternalSystem catalog 参照 (context.catalogs.externalSystems)
+ * - ext    : @ext.<namespace> Extension namespace 参照
  *
  * @env.* は special-case として checkStep 内で context.catalogs.envVars と突合するため
  * ここには含めない (下記 root === "env" ブランチを参照)。
@@ -76,9 +81,10 @@ const BUILTIN_AMBIENTS = new Set<string>([
   "secret",
   "conv",
   "ambient",
-  // Generic Definition Catalog refs (#1285)
+  // Generic Definition Catalog refs (#1285、14 kind 全件)
   "msg",
   "validation",
+  "rule",
   "const",
   "event",
   "logEvent",
@@ -90,12 +96,16 @@ const BUILTIN_AMBIENTS = new Set<string>([
   "exception",
   "policy",
   "component",
-  // Top-level entity hierarchical refs (#1285)
+  // Top-level entity / catalog refs (#1285)
   "screen",
   "table",
   "view",
   "viewer",
   "layout",
+  "seq",
+  "flow",
+  "system",
+  "ext",
 ]);
 
 /** 任意の文字列から @identifier と property path を抽出 */

--- a/frontend/src/schemas/identifierScope.ts
+++ b/frontend/src/schemas/identifierScope.ts
@@ -33,6 +33,7 @@ export interface IdentifierIssue {
  * 組み込み関数・グローバル識別子。
  * これらは宣言なしで常に参照可能なため、スコープ検査から除外する。
  *
+ * **組み込み関数 / グローバル**:
  * - fn    : @fn.calcXxx(...) 形式の業務関数呼び出し
  * - now   : @now  現在時刻 (Timestamp)
  * - uuid  : @uuid 新規 UUID 生成
@@ -40,16 +41,61 @@ export interface IdentifierIssue {
  * - conv  : @conv.* conventions 参照 (conventionsValidator でカバー)
  * - ambient: @ambient.* 旧形式の ambient 参照
  *
+ * **Generic Definition Catalog 参照** (docs/spec/process-flow-prefix-system.md §3、
+ * #1285 で追加): broken-ref 検証は processFlowAntipatternValidator Check 31 が担うため
+ * identifierScope は変数 scope 検査の対象外として skip する。
+ * - msg        : @msg.<Name> generic-definitions/message
+ * - validation : @validation.<Name> generic-definitions/validation-rule
+ * - const      : @const.<Name>.<key> generic-definitions/constants
+ * - event      : @event.<topic> generic-definitions/domain-event (catalogs.events と兼用)
+ * - logEvent   : @logEvent.<Name> generic-definitions/log-event
+ * - logConfig  : @logConfig.<Name> generic-definitions/log-config
+ * - fragment   : @fragment.<Name> generic-definitions/ui-fragment
+ * - behavior   : @behavior.<Name> generic-definitions/ui-behavior
+ * - contract   : @contract.<Name> generic-definitions/data-contract
+ * - type       : @type.<Name> generic-definitions/domain-type
+ * - exception  : @exception.<Name> generic-definitions/exception-type
+ * - policy     : @policy.<Name> generic-definitions/runtime-policy
+ * - component  : @component.<Name> generic-definitions/component-definition
+ *
+ * **Top-level entity 階層参照** (process-flow-prefix-system.md §3、#1285 で追加):
+ * - screen : @screen.<id>.item.<id>.<field> Screen entity 参照
+ * - table  : @table.<id>.field.<id>.<field> Table entity 参照
+ * - view   : @view.<id>.field.<col> View entity 参照
+ * - viewer : @viewer.<id> ViewDefinition 参照
+ * - layout : @layout.<id> PageLayout 参照
+ *
  * @env.* は special-case として checkStep 内で context.catalogs.envVars と突合するため
  * ここには含めない (下記 root === "env" ブランチを参照)。
  */
 const BUILTIN_AMBIENTS = new Set<string>([
+  // Built-in functions / globals
   "fn",
   "now",
   "uuid",
   "secret",
   "conv",
   "ambient",
+  // Generic Definition Catalog refs (#1285)
+  "msg",
+  "validation",
+  "const",
+  "event",
+  "logEvent",
+  "logConfig",
+  "fragment",
+  "behavior",
+  "contract",
+  "type",
+  "exception",
+  "policy",
+  "component",
+  // Top-level entity hierarchical refs (#1285)
+  "screen",
+  "table",
+  "view",
+  "viewer",
+  "layout",
 ]);
 
 /** 任意の文字列から @identifier と property path を抽出 */

